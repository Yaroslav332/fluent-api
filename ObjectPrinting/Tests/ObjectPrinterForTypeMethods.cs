using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace ObjectPrinting.Tests
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ObjectPrinterForTypeMethods
    {
        [Test]
        public void ExcludeType_Should_Exclude()
        {
            var p = new Person() {Name = "Alex", Height = 35.5, Weight = 83.3};
            var printer = ObjectPrinter.For<Person>()
                .ExcludeType<double>();
            Approvals.Verify(printer.PrintToString(p));
        }

        [Test]
        public void AlternateType_Should_Alternate()
        {
            var p = new Person() {Name = "Alex", Height = 35.5, Weight = 83.3};
            var printer = ObjectPrinter.For<Person>()
                .SetSerializeMethodForType<double>(d => d + " hehe");
            Approvals.Verify(printer.PrintToString(p));
        }
    }
}