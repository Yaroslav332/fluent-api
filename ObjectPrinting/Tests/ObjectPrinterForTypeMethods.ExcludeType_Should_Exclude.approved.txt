Person
	Id = Guid
		Empty = !Cyclic_Reference!
	Name = Alex
	Birthday = 01.01.0001 0:00:00
	RunRatio = 0
	Age = 0
	RunSpeed = 0
	Parent = null
